src/utilities.cpp

```diff
@@ -20,27 +20,15 @@ limitations under the License.
 #include <iomanip>
 #include <limits>
 #include <set>
+#include <vector>
+#include <iostream>
 #include <sstream>
-#include <stdexcept>
 
 namespace libcellml {
 
-bool convertToDouble(const std::string &candidate, double *value)
+double convertToDouble(const std::string &candidate)
 {
-    bool canConvert = false;
-    // Try to convert the candidate string to double.
-    try
-    {
-        double tmp = std::stod(candidate);
-        if (value) {
-            *value = tmp;
-        }
-        canConvert = true;
-    } catch (...) {
-        canConvert = false;
-    }
-
-    return canConvert;
+    return std::stod(candidate);
```

Comment:
Is this guaranteed to go right?

Reply:
That's the idea, the candidate is first check for conformance against the spec. so anything that passes that should not cause any problems for stod.  That's the theory anyway (yet to be proven false).

Reply:
Oh no, you're right what is expected when we come across something like this `5.0E+310`?

Reply:
No idea! @agarny @nickerso do you remember if MathML specifies any specific float format? e310 won't be allowed in lots of ISO formats and such

Reply:
I can't find anything in the cellml or mathml spec that disallows this. If we were going ultra-purist we could keep the number as a string, and let the code _using_ libcellml decide what size its floats can be. However, for now I'd prefer it if we just added **a non-spec error** saying we found a float we can't parse

Reply:
We can make it better if someone actually wants it. Don't see that happening soon